src/averageAges.js

```diff
@@ -15,7 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
+  const onlyMen = people.filter(person => person.sex === 'm');
+  const men = century
+    ? onlyMen.filter(person => Math.ceil(person.died / 100) === century)
+    : onlyMen;
+  const averageAge = men.reduce((sum, i) => sum + (i.died - i.born), 0);
+
+  return averageAge / men.length;
```

Comment:
move logic of calculating an average age to the separate function so you will be able to reuse it in your function instead of duplicate the same code